Remove COLOURLovers UI features

abotic opened the pull request:
Fixes DOTCOM-1757

## Changes Overview

* Remove all COLOURlovers UI elements from the color customizer 
* Maintain display of existing COLOURlovers patterns/palettes for sites that currently use them
* Remove ability to add new or update existing COLOURlovers patterns/palettes for all users

## Why are these changes being made?

* We are ending a contract with COLOURlovers described in linked P2 post
* The integration has been partially non-functional for some time (patterns endpoint has been broken since 2019)
* The service is primarily used by legacy (non-block) themes in the customizer

## Jetpack product discussion

Related to pdKhl6-3qP-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. **Environment Setup**
   * Apply these changes to your sandbox
   * Create or select a atomic WordPress.com site
   * Activate the Twenty Seventeen theme (wordpress.com/theme/twentyseventeen)
   * Sandbox the site

2. **Verify UI Removal** 
   * Navigate to WP Admin > Appearance > Customize (/wp-admin/customize.php)
   * Click on "Colors & Backgrounds"
   * Verify that:
     - No COLOURlovers palette/pattern selection UI is visible
     - No ability to add new COLOURlovers patterns/palettes
     - Core color picker functionality works normally
     - Basic color customization options are available

3. **Legacy Pattern/Palette Check**
   * On a site with existing COLOURlovers patterns/palettes:
     - Verify existing patterns/palettes continue displaying correctly
     - Confirm that there is no UI to modify existing COLOURlovers patterns/palettes
     - Verify other color customization features work properly

## Expected Results

* No users should see COLOURlovers UI or have ability to add/update ColourLovers patterns/palettes
* Sites with existing COLOURlovers patterns/palettes should continue displaying them correctly
* Core color customization features should work without interruption

|Before|After|
|---|---|
|<img width="310" alt="image" src="https://github.com/user-attachments/assets/4d16df3c-f797-4cd2-a9f4-46264f899017" />|<img width="307" alt="image" src="https://github.com/user-attachments/assets/ca4c5c79-bd8d-4979-ad9c-b52f1881ddba" />|

abotic committed:
Remove COLOURLovers UI features

abotic committed:
changelog

abotic committed:
remove unused css

abotic committed:
remove unused js

xavier-lc commented on projects/plugins/wpcomsh/custom-colors/js/colors-control.js:
```diff
@@ -547,7 +547,6 @@
 						$( this ).hasClass( 'bg-change-disable' ) === false
 					) {
 						ct.backgroundChangeView.open();
-						$( '#colourlovers-palettes-container' ).hide();
```
There's more JS that can be removed, like `initPalettes`/`initPatterns`

abotic replied:
They both works differently than simple sites, and while there are things that we could potentially remove - in all methods there is a lot of code that is not connected to COLOURlovers functionality.

In this PR, I have removed the UI for them, and when we end contract, we can remove the remaining things.

dsas replied:
We should make sure that there are follow-up issues for any work to be done afterwards, otherwise it'll get forgotten.

xavier-lc reviewed (dismissed):
Works as described 👌

![image](https://github.com/user-attachments/assets/ab76e9f2-27b9-44b4-bad0-713d5d844817)

Copons committed:
Fix lint and remove more CL rules

Copons committed:
Remove COLOURLovers palette and pattern JS code

Mirrors the JS cleanup in Automattic/wpcom#172119: drops the
palette/pattern UI plumbing (initPalettes, initPatterns,
showPalettes, showPatterns, colorPalettes, colorPatterns,
handleMorePatternsClick, hideSectionAndKeepHidden,
generatePaletteFromHeader, showBackgroundColorChangeOptions,
showBackgroundPatternOptions) and the AJAX calls to color_palettes,
pattern_recommendations, and generate_palette. Without this, the
customizer kept hitting the (defunct) COLOURLovers backend on every
load even though the UI was gone.
